packages/html-reporter/src/headerView.tsx

```diff
@@ -31,12 +31,25 @@ export const HeaderView: React.FC<React.PropsWithChildren<{
   projectNames: string[],
 }>> = ({ stats, filterText, setFilterText, projectNames }) => {
   React.useEffect(() => {
-    (async () => {
-      window.addEventListener('popstate', () => {
-        const params = new URLSearchParams(window.location.hash.slice(1));
-        setFilterText(params.get('q') || '');
-      });
-    })();
+    const popstateFn = () => {
```

Comment:
Is this necessary? Looks like this code will end up in the Link/Router infrastructure rather than scattered across the components. Let's do one thing at a time.

Reply:
i'm removed inputFn with his listeners but left `popstateFn` with added `removeEventListener` to avoiding duplication listerers

old implementation 

https://user-images.githubusercontent.com/11705902/224674539-39413e69-9184-44d4-aaf2-a596290d00f1.mp4

new

https://user-images.githubusercontent.com/11705902/224674669-18f4e15e-df5e-48f0-b039-a3701fda5a45.mp4